addons/l10n_pt_account/models/account_move.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api
+
+
+class AccountMove(models.Model):
+    _name = 'account.move'
+    _inherit = ['account.move', 'l10n_pt.mixin']
+
+    # Override l10n_pt.mixin
+    @api.depends('company_id.account_fiscal_country_id.code', 'move_type', 'sequence_prefix', 'sequence_number')
+    def _compute_l10n_pt_document_number(self):
+        for move in self.filtered(lambda m: (
+            m.company_id.account_fiscal_country_id.code == 'PT'
+            and m.move_type in ('out_invoice', 'out_refund', 'in_invoice', 'in_refund')
+            and m.sequence_prefix
+            and m.sequence_number
+        )):
+            seq_info = self._l10n_pt_get_sanitized_sequence_info(move.sequence_prefix, number=move.sequence_number)
+            move.write({'l10n_pt_document_number': f"{move.move_type} {seq_info}"})
```

Comment:
Using write disables optimizations of the ORM, you could also add the dependencies or `_compute_l10n_pt_document_number` in the list of fields that triggers the warning in the `write` instead of adding `l10n_pt_document_number`